core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -450,14 +430,12 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       newMetadataTmp.configuration.contains(Protocol.MIN_WRITER_VERSION_PROP)) {
       // Collect new reader and writer versions from table properties, which could be provided by
       // the user in `ALTER TABLE TBLPROPERTIES` or copied over from session defaults.
-      val readerVersionInNewMetadataTmp = newMetadataTmp.configuration
-        .get(Protocol.MIN_READER_VERSION_PROP)
-        .map(Protocol.getVersion(Protocol.MIN_READER_VERSION_PROP, _))
-        .getOrElse(protocolBeforeUpdate.minReaderVersion)
-      val writerVersionInNewMetadataTmp = newMetadataTmp.configuration
-        .get(Protocol.MIN_WRITER_VERSION_PROP)
-        .map(Protocol.getVersion(Protocol.MIN_WRITER_VERSION_PROP, _))
-        .getOrElse(protocolBeforeUpdate.minWriterVersion)
+      val (readerVersionInNewMetadataTmp, writerVersionInNewMetadataTmp) = {
+        val (readerVersionInNewMetadataTmpOpt, writerVersionInNewMetadataTmpOpt) =
+          Protocol.getProtocolVersionsFromTableConf(newMetadataTmp.configuration)
+        (readerVersionInNewMetadataTmpOpt.getOrElse(protocolBeforeUpdate.minReaderVersion),
+          writerVersionInNewMetadataTmpOpt.getOrElse(protocolBeforeUpdate.minWriterVersion))
+      }
```

Comment:
nit: This is a bit hard to read. How about
```suggestion
      val (readerVersionInNewMetadataTmpOpt, writerVersionInNewMetadataTmpOpt) =
          Protocol.getProtocolVersionsFromTableConf(newMetadataTmp.configuration)
      val (readerVersionInNewMetadataTmp, writerVersionInNewMetadataTmp) =
        (readerVersionInNewMetadataTmpOpt.getOrElse(protocolBeforeUpdate.minReaderVersion),
          writerVersionInNewMetadataTmpOpt.getOrElse(protocolBeforeUpdate.minWriterVersion))
```
or maybe add a helper to `Protocol` that returns the protocol version for the reader and writer separately? So then we can do 
```
val readerVersionInNewMetadataTmp = Protocol.getReaderVersionsFromTableConf(newMetadataTmp.configuration).getOrElse(protocolBeforeUpdate.minReaderVersion)
```

Reply:
Added!